tests/Feature/Sensors/JobAttemptSensorTest.php

```diff
@@ -51,17 +67,55 @@ protected function setUp(): void
         Redis::command('FLUSHALL');
     }
 
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        if ($this->isVapor) {
+            putenv('VAPOR_SSM_PATH');
+
+            $reflection = new ReflectionClass(VaporWorkCommand::class);
+            $property = $reflection->getProperty('listeningForEvents');
+            $property->setAccessible(true);
+            $property->setValue(null, false);
```

Comment:
`VaporWorkCommand` registers event listeners if the static property `listeningForEvents` is false.
Without manually resetting it to false, the event listeners won't be registered, and the command will not insert failed jobs into the database.
https://github.com/laravel/vapor-core/blob/2.0/src/Console/Commands/VaporWorkCommand.php#L48-L53
https://github.com/laravel/vapor-core/blob/2.0/src/Console/Commands/WritesQueueEventMessages.php

The framework's `WorkCommand` provides the `flushState` method, but `VaporWorkCommand` doesn't. So, I had to reset the protected value using reflection.

Reply:
Opened a PR in vapor-core: https://github.com/laravel/vapor-core/pull/191